### DISCUSSION TITLE
First experience of migrating development from windows to MAC Apple Silicon M1.

### DISCUSSION THREAD

**Discussion by PavelVsl** [oqtane/oqtane.framework#2298] [Show and tell]
One fine day I thought that life is a change and I bought a brand new MAC Studio M1. Without much experience with MAC computers, I embarked on a new adventure.

After a week of intense research and struggling with an unfamiliar environment, I finally saw success. I am able to run, debug and publish my application based on Oqtane framework.

Resulting software configuration:

- Rider 
- Docker
- Azure SQL Edge in Docker for database storage
- Smtp4Dev docker image for E-mail catching (https://github.com/rnwood/smtp4dev)
- some shell scripts for replication of production database to local SQL and for publishing result to IIS server.
- SQL script for migration of production database to development environment 

Problems I had to solve:
1. I had to replace the non-existent Sql server for MAC using the Azure SQL Edge database in Docker.
1. Alternative configurations (appsettings.Development.json) did not work (#2295)
2. DirectorySeparator issues (discussed #908, not fully resolved). After transferring the database from Windows environment to Unix, I have to replace all "\" with "/" using SQL script.
3. Unhandled error detected in case of non-existent folder. (#2296)
4. The application crashes on start if the Modules, Themes and Packages directories are missing from the site root.(#2299)

After solving these minor problems, I must state that Oqtane can be developed on MAC M1.

**Comment by sbwalker** [comment]
@chlupac can you please explain the scenario which is causing "4. Unhandled error detected in case of non-existent folder" - ie. what are the steps to reproduce, as I do not understand how this issue could be caused by running on a Mac?

**Comment by PavelVsl** [comment]
It's simple, it happens before I fix Path property of folders. Folder name in database was `"Documents\" `and module was trying to open folder `"Documents/" ` and  `return files.OrderBy(item => item.Name).ToList();` thhrow exception because files is null.

**Reply by sbwalker** [reply]
Thank you for explaining... this makes sense. I assume this means that the framework should be modified to store Folder Path in a cross platform format. Were there any other database fields which needed to be modified?

**Reply by PavelVsl** [reply]
No, I do not have other problems.

**Comment by sbwalker** [comment]
Is there a reason why you decided to use Azure SQL Edge database in Docker rather than one of the other cross platform database providers such as SQLite, PostgreSQL, or MySQL (all of which run on a Mac)?

**Reply by PavelVsl** [reply]
Because it is SQL server compatible database and production server is running on IIS with SQL server. I can easily restore backup from production.
